### ISSUE TITLE
TODO: Wiki page for the modules frontend conventions

### ISSUE THREAD

**Issue by edhelas** [edhelas/movim#132]
For the Roster module in particular, and for other modules in the future I guess as well, some user preference informations are stored in the localStorage of the browser.

These MUST NOT be sensitive information since anybody can access them by the browser console.
Nevertheless, this localStorage is shared by any user that would connect to Movim with the same browser of the same computer and it would be convenient to have a separation between the data of different users.

The convention for now would be to create one variable by user and by module with the name <jid>_<ModuleName> (pay attention to the capitalization!). In my case it would be <jid>_Roster.

In the case of the Roster module, these localStorage variables are mostly used for caching, which means I need to delete my cache variables at the end of the session (when the user logs out or when the session breaks). Therefore, a good place to manage the localStorage would be the login page (thank you edhelas!).

The case of deleting cache variables while keeping other configuration variables like the state of the groups remains undecided... Maybe have a localStorage variable for all cache of all modules ?  Maybe <jid>_cache (this time with a lowercase C in case one day we have a Cache module)?

This way I could just dump the whole <jid>_cache in the login page (keeping the jid in the variable name would be the "better be safe than sorry" detail to not grab the cache of another user...).

TL;DR:
DON'T put sensitive info in your localStorage.
If you put something in it, follow the structures below.

Configuration for group state you want to keep between sessions:
<jid>_Roster{
    <variableName>: <any value>,
    ...
}

Cache variables you want to kep inside the session only:
<jid>_cache{
    <ModuleName>: {
        <variableName>: <any value>,
        ...
    }
    ...
}

... Or feel free to explain why you disagree \o/

[EDIT]: Implementation showed that '@' is not allowed in localStorage key. I replaced it by 'at'.
